{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0a8afc86",
   "metadata": {},
   "source": [
    "# Natural Selection\n",
    "\n",
    "# Natural Selection \n",
    "\n",
    "Natural selection is arguably the single most important evolutionary\n",
    "force. The products of natural selection surround us- our biochemical\n",
    "sophistication, our finely tuned genetic systems, the incredible fit of\n",
    "organisms to their environments- however witnessing the change brought\n",
    "about by natural selection is nearly impossible in the human life time.\n",
    "Evolutionary change is simple too slow for the human observer. Thus we\n",
    "turn our attention to mathematical models of natural selection which\n",
    "when poked and prodded, will reveal much of the subtle nuance of\n",
    "evolution via natural selection.\n",
    "\n",
    "# The basic model of viability selection \n",
    "\n",
    "We will again turn to our simplistic model of diploid, randomly mating\n",
    "hermaphrodites to study the fundementals of natural selection. Keep in\n",
    "mind that this model is for convenience, we can also study more\n",
    "realistic models (dioecious species, non-random mating, etc.), however\n",
    "our basic model has nearly all of the same features as more realistic\n",
    "models and is readily tackled mathematically.\n",
    "\n",
    "Our little hermaphrodites have a simple life cycle, randomly chosen\n",
    "gametes fuse to form zygotes, then these zygotes must survive to\n",
    "adulthood, adults must then be fertile to reproduce and produce\n",
    "offspring. You'll notice that we've added two parts to our simple\n",
    "Hardy-Weinberg model, namely survivorship and fecundity. If the $A_iA_i$\n",
    "genotype has genotype-specific survivorship $l_{ii}$ and fecundity\n",
    "$m_{ii}$, then the life cycle outlined above can be represented as\n",
    "Genotype $\\rightarrow$ zygote $\\rightarrow$ $l_{ii}\\rightarrow$ adult\n",
    "$\\rightarrow$ $m_{ii} \\rightarrow$ offspring. We are now ready to\n",
    "understand how survivorship and fecundity differences between genotypes\n",
    "can lead to differences in genotype frequencies, outlined in Table 1.\n",
    "\n",
    "```{table} \n",
    "   |Genotype $\\rightarrow$  | zygote $\\rightarrow$ |  $l_{ii}\\rightarrow$|   adult $\\rightarrow$|   $m_{ii} \\rightarrow$|       offspring |\n",
    "   |------------------------|----------------------|---------------------|---------------------|----------------------|-------------------|\n",
    "   |       $A_1A_1$         |       $p^2$          |                     |    $p^2l_{11}$      |                      | $p^2m_{11}l_{11}$ |\n",
    "   |       $A_1A_2$         |       $2pq$          |                     |    $2pql_{12}$      |                      | $2pqm_{12}l_{12}$ |\n",
    "   |       $A_2A_2$         |       $q^2$          |                     |    $q^2l_{22}$      |                      | $q^2m_{22}l_{22}$ |\n",
    "```\n",
    "\n",
    "For the most part, we will consider the probability of representation to\n",
    "be the joint probability of surviving to adulthood and reproducing. This\n",
    "leads to our first formal definition of viability or absolute fitness.\n",
    "Let the viabilities of the $A_1A_1$, $A_1A_2$, and $A_2A_2$ genotypes be\n",
    "$w_{11}$,$w_{12}$, and $w_{22}$ respectively. Further let these\n",
    "viabilities be the joint probability of survival and reproduction such\n",
    "that $w_{ii} = l_{ii} \\times m_{ii}$. Thus we can see that the frequency\n",
    "of a genotype after a round of selection (i.e. in the next generation\n",
    "after selection has occured) is proportional to the frequency of a\n",
    "genotype before selection times the viability of that genotype. For\n",
    "example, the frequency of the $A_1A_2$ genotype after a round of\n",
    "selection is proportional $2pqw_{12}$.\n",
    "\n",
    "Next we are interested in figuring out the *relative* frequencies of\n",
    "genotypes after a round of selection. To do this we require some\n",
    "constant that will allow genotype frequencies after selection to sum to\n",
    "one. This is where the average fitness of a population comes in as we\n",
    "looked at in lecture.\n",
    "\n",
    "```{table}\n",
    "|  Genotype:                    |       $A_1A_1$      |        $A_1A_2$     |         $A_2A_2$\n",
    "|-----------------------------|---------------------|---------------------|---------------------|\n",
    "| Frequency before selection: |          $p^2$      |           $2pq$     |            $q^2$    |\n",
    "| Viability:                  |       $w_{11}$      |       $w_{12}$      |        $w_{22}$      |\n",
    "| Frequency after selection:  |   $p^2w_{11}/\\bar{w}$ |   $2pqw_{12}/\\bar{w}$ |   $q^2w_{22}/\\bar{w}$ |\n",
    "  \n",
    "```\n",
    "\n",
    "The constant, $\\bar{w}$, is equal to \n",
    "\n",
    "$$\\begin{aligned}\n",
    "    \\bar{w} = p^2w_{11} + 2pqw_{12} + q^2w_{22}.\n",
    "\\end{aligned}$$\n",
    "\n",
    "First notice that $\\bar{w}$ represents the frequency weighted average fitness\n",
    "of our population. Next also note that this didn't pop out of thin air\n",
    "but is instead chosen such that \n",
    "\n",
    "$$\\begin{aligned}\n",
    "    \\frac{p^2w_{11}}{\\bar{w}} + \\frac{2pqw_{12}}{\\bar{w}} + \\frac{q^2w_{22}}{\\bar{w}} = 1\n",
    "\\end{aligned}$$\n",
    "\n",
    "thus making sure that we can calculate relative frequencies of genotypes\n",
    "after selection. Examine this equation and the table closely- what\n",
    "you'll notice is that what is important in determining the frequency of\n",
    "a genotype in the next generation is not it's viability *per se*, but\n",
    "instead its viability *relative* to the average viability (fitness) of\n",
    "the population.\n",
    "\n",
    "Unlike under Hardy-Weinberg dynamics, natural selection can change\n",
    "allele frequencies. If we focus attention on the $A_1$ allele we can\n",
    "write down its frequency after a round of selection $p'$ as\n",
    "\n",
    "$$\\begin{aligned}\n",
    "    p' = \\frac{p^2w_{11} + pq{w_{12}}}{\\bar{w}}.\n",
    "\\end{aligned}$$ \n",
    "\n",
    "Ask yourself this question: why is the second term in the numerator\n",
    "$pqw_{12}$ and not $2pqw_{12}$?\n",
    "\n",
    "```{admonition} Exercise\n",
    "write a function to compute $p'$ from some input $p$\n",
    "and some set of fitness values, $w_{ii}s$.\n",
    "Next, use this function to plot how selection will change allele \n",
    "allele frequencies over time. \n",
    "```\n",
    "\n",
    "The next order of business is to write down the change in allele\n",
    "frequency due to selection in a single generation. That is\n",
    "$\\Delta_sp = p' - p$, \n",
    "\n",
    "$$\\begin{aligned}\n",
    "    p' - p & = & \\frac{p^2w_{11} + pq{w_{12}} - p\\bar{w}}{\\bar{w}} \\\\\n",
    "           & =  & \\frac{p[pqw_{11} + q(1 - 2p)w_{12} - q^2w_{22}]}{\\bar{w}},\n",
    "\\end{aligned}$$\n",
    "\n",
    "which after some algebraic fiddling yields\n",
    "\n",
    "$$\\begin{aligned}\n",
    "    \\Delta_sp = \\frac{pq[p(w_{11} - w_{12}) + q(w_{12} - w_{22})]}{\\bar{w}}\n",
    "\\end{aligned}$$\n",
    "\n",
    "We can now see how selection alone should change allele frequencies as a\n",
    "function of genotypic viability. This result is certainly one of the\n",
    "most important results in all of population genetics and evolution.\n",
    "\n",
    "# Relative Fitness\n",
    "\n",
    "As we noticed earlier and made plain in equation 6, the change in allele\n",
    "frequency due to selection is dependent not just on the viability of a\n",
    "genotype alone, but instead on the viability of that genotype relative\n",
    "to the average viability of the population. This leads us to the\n",
    "distinction between absolute fitness and relative fitness. Absolute\n",
    "fitness can be understood as the probability of a genotype surviving to\n",
    "adulthood and then reproducing. That's all well and good, but when we\n",
    "are interested in evolution the only thing that counts is how well that\n",
    "genotype performs in relation to others in its population. Thus we will\n",
    "define relative fitness to be the viability of a genotype standardized\n",
    "by the maximum genotypic viability. This won't change any of the\n",
    "mechanics that we just derived, but instead will make them clearer to\n",
    "understand. Consider the following table\n",
    "\n",
    "```{table}\n",
    "  |  Genotype:                    |       $A_1A_1$      |        $A_1A_2$     |         $A_2A_2$\n",
    " |-----------------------------|---------------------|---------------------|---------------------|\n",
    " | Viability:            | $w_{11}$ |      $w_{12}$ |          $w_{22}$ |\n",
    " | Relative viability:    |   1       | $w_{12}/w_{11}$  | $w_{22}/w_{11}$ |\n",
    "  \n",
    "```\n",
    "\n",
    "So far we've been calling the $w_{ij}$'s viabilities- in the business\n",
    "they are more commonly referred to as the absolute fitnesses of the\n",
    "individual genotypes. One we have normalized the $w_{ij}$'s by the most\n",
    "fit genotype in the population we end up with relative fitness values\n",
    "for each genotype. The common notation for relative fitness is\n",
    "\n",
    "```{table}\n",
    "\n",
    "  |  Genotype:            |       $A_1A_1$      |        $A_1A_2$     |         $A_2A_2$\n",
    "  |-----------------------|---------------------|---------------------|---------------------|\n",
    "  | Relative Fitness:     | $1$                 |   $1 -hs$           |  $1 - s$            |\n",
    "```\n",
    "\n",
    "where $1-hs = w_{12}/w_{11}$ and $1-s = w_{22}/w_{11}$. The parameter\n",
    "$s$ is called the selection coefficient. It is a measure of how fit the\n",
    "$A_2A_2$ genotype is relative to the $A_1A_1$ genotype. In this setting,\n",
    "if the selection coefficient is positive then the $A_2A_2$ genotype is\n",
    "less fit than the $A_1A_1$ genotype. Conversely if the selection\n",
    "coefficient is negative, then the $A_2A_2$ genotype is more fit. We'll\n",
    "always assume that $0\\leq s \\leq1$ in this class, but notice that all of\n",
    "this notation is really just arbitrary. We're simply assuming that the\n",
    "$A_1$ allele is generally the confers some fitness benefit to the\n",
    "genotype.\n",
    "\n",
    "What about the $h$ parameter? This parameter is sometimes called the\n",
    "heterozygous effect, as it measures the difference in fitness between\n",
    "the heterozygotes and the two homozygote genotypes. In essence though,\n",
    "$h$ really defines the dominance relations between the 2 alleles at our\n",
    "locus. This is shown in the following table\n",
    "\n",
    "```{table} Table showing dominance relationships among alleles specified by $h$\n",
    "  |  $h$        |    dominance |\n",
    "  |-------------|--------------------------------|\n",
    "  |   $h = 0$   |  $A_1$ dominant, $A_2$ recessive |\n",
    "  |   $h = 1$   |  $A_2$ dominant, $A_1$ recessive |\n",
    "  |   $0 < h < 1$ |           co-dominance            |\n",
    "  |   $h < 0$     |         overdominance |\n",
    "  |   $h > 1$     |        underdominance |\n",
    "```\n",
    "\n",
    "Generally in genetics we are interested in the cases of co-dominance,\n",
    "overdominance, and underdominance. The complete dominance setting,\n",
    "although seen in morphological traits often, is generally unrealistic\n",
    "for most genetic situations. Why would this be?\n",
    "\n",
    "We are now fully armed to rewrite equation 6 in terms of relative\n",
    "fitness. This will lead to a much more provocative construction of this\n",
    "same equation. So the change in frequency of the $A_1$ allele due to\n",
    "selection is then \n",
    "\n",
    "$$\\begin{aligned}\n",
    "    \\Delta_sp = \\frac{pqs[ph + q(1-h)]}{\\bar{w}}\n",
    "\\end{aligned}$$\n",
    "\n",
    "and we can write down the population mean fitness as \n",
    "\n",
    "$$\\begin{aligned}\n",
    "    \\bar{w} = 1 - 2pqhs - q^2s.\n",
    "\\end{aligned}$$ \n",
    "\n",
    "The only thing I've done\n",
    "to get these formulations is replace $w_{11}, w_{12}$ and $w_{22}$ with\n",
    "$1,1-hs$, and $1-s$ respectively.\n",
    "\n",
    "\n",
    "```{admonition} Exercise\n",
    ":class: note\n",
    "\n",
    "rewrite your function from above to compute $p'$ \n",
    "from some input $p$ and a value of $h$ and $s$\n",
    "Next, use this function to plot how selection will change allele \n",
    "allele frequencies over time. \n",
    "\n",
    "```\n",
    "\n",
    "# Flavors of selection \n",
    "\n",
    "We talk generally about 3 kinds of selection: directional selection (as\n",
    "occurs with co-dominance), balancing selection (overdominance), and\n",
    "disruptive selection (underdominance). We have already looked at what\n",
    "these types of selection do at the phenotypic level during lecture. What\n",
    "are the effects of these distinct flavors of selection on allele\n",
    "frequencies? To examine this we will examine the fate of the $A_1$\n",
    "allele in a population, given values for the parameters $h$ and $s$ and\n",
    "assuming some initial frequency of the $A_1$ allele.\n",
    "\n",
    "## Directional selection \n",
    "\n",
    "Directional selection is perhaps the most familiar form of selection.\n",
    "This is the kind of selection that Charles Darwin had in mind when he\n",
    "wrote about evolution by natural selection. Directional selection occurs\n",
    "in the case of incomplete dominance ($0 \\leq h \\leq 1$). In this case\n",
    "the frequency of the $A_1$ allele, $p$, should always increase, or seen\n",
    "another way $\\Delta_sp > 0$. {numref}`dir_seln` plots $\\Delta_sp$ as a function of\n",
    "$p$. As an exercise, plot how $p$ changes from generation to generation.\n",
    "Does the $A_1$ allele always fix (i.e. $\\rightarrow 1$) in the\n",
    "population?\n",
    "\n",
    "\n",
    "```{figure} figures/dir_seln_delta.png\n",
    "---\n",
    "name: dir_seln\n",
    "---\n",
    "Directional selection. Shown is the change in allele frequency\n",
    "$\\Delta_sp$ as a function of allele frequency $p$. In this case $h=0.5$\n",
    "and $s=0.1$\n",
    "```\n",
    "\n",
    "\n",
    "One thing to notice- the change in allele frequency is slowest when\n",
    "there is little genetic variation in the population (i.e. when $p$ is\n",
    "close to zero or one). Be sure that you can explain to yourself why this\n",
    "is.\n",
    "\n",
    "## Balancing selection \n",
    "\n",
    "The second kind of selection occurs when there is overdominance (i.e.\n",
    "$h < 0$). Unlike directional selection, balancing selection leads to the\n",
    "maintenance of variation within a population. That is to say the the\n",
    "allele frequency of the $A_1$ allele will approach some equilibrium\n",
    "value $\\hat{p}$ independent of initial frequency. We will plot this\n",
    "again graphically by looking at the change in allele frequency as a\n",
    "function of allele frequency.\n",
    "\n",
    "\n",
    "```{figure} figures/bal_seln_delta.png\n",
    "---\n",
    "name: bal_seln\n",
    "---\n",
    "Balancing selection. Shown is the change in allele frequency\n",
    "$\\Delta_sp$ as a function of allele frequency $p$. In this case $h=-0.5$\n",
    "and $s=0.1$\n",
    "```\n",
    ".\n",
    "\n",
    "Let's look closely at {numref}`bal_seln`. Notice that when $p$ is close to zero\n",
    "$\\Delta_sp > 0$ and thus allele frequency will increase when rare.\n",
    "Conversely when $p$ is close to one $\\Delta_sp < 0$ and allele frequency\n",
    "will decrease. Where is the stable equilibrium in this figure? It is\n",
    "left to the reader to find this equilibrium point. You can do so by\n",
    "setting $\\Delta_sp = 0$ and solving for $\\hat{p}$.\n",
    "\n",
    "## Disruptive selection\n",
    "\n",
    "The last kind of selection we will look at is disruptive selection. This\n",
    "occurs with overdominance, $h > 1$. {numref}`disrup_seln` again graphs $\\Delta_sp$ as\n",
    "a function of $p$. What's curious in this case is that $p$ will decrease\n",
    "when rare and increase when common in the population. This means that if\n",
    "$p$ is less than $\\hat{p}$ the allele will be lost, and if $p>\\hat{p}$\n",
    "it will fix. Thus $\\hat{p}$ is an unstable equilibrium. If by some\n",
    "strange circumstance $p = \\hat{p}$ allele frequencies won't change at\n",
    "all! Needless to say underdominance is rare in nature\\...\n",
    "\n",
    "\n",
    "```{figure} figures/disrup_seln_delta.png\n",
    "---\n",
    "name: disrup_seln\n",
    "---\n",
    "**Disruptive selection.** Shown is the change in allele frequency\n",
    "$\\Delta_sp$ as a function of allele frequency $p$. In this case $h=2$\n",
    "and $s=0.1$\n",
    "```\n",
    "."
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "formats": "md:myst",
   "text_representation": {
    "extension": ".md",
    "format_name": "myst",
    "format_version": 0.13,
    "jupytext_version": "1.11.5"
   }
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.9"
  },
  "source_map": [
   13
  ]
 },
 "nbformat": 4,
 "nbformat_minor": 5
}