
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Migration and Gene Flow &#8212; Bio 610-- Intro to Population Genetics</title>
    
  <link href="_static/css/theme.css" rel="stylesheet">
  <link href="_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script>
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Molecular Evolution" href="lecture5.html" />
    <link rel="prev" title="Mutation" href="lecture3.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/tree.jpeg" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Bio 610-- Intro to Population Genetics</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Welcome to Population Genetics
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Population Genetics Concepts
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="lecture1.html">
   Introduction to Population Genetics
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture2.html">
   Natural Selection
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture3.html">
   Mutation
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Migration and Gene Flow
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture5.html">
   Molecular Evolution
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_drift.html">
   Genetic drift
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_diversity.html">
   Genetic diversity, with trees
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Exercises
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="exercise1.html">
   Exercise 1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="exercise2.html">
   Exercise 2
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/lecture4.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/kr-colab/bio610_popgen"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/kr-colab/bio610_popgen/issues/new?title=Issue%20on%20page%20%2Flecture4.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/kr-colab/bio610_popgen/main?urlpath=lab/tree/book/lecture4.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#non-random-mating">
   Non-random mating
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#generalized-hardy-weinberg">
   Generalized Hardy-Weinberg
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#slim-simulations-with-migration">
     SLiM simulations with migration
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Migration and Gene Flow</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#non-random-mating">
   Non-random mating
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#generalized-hardy-weinberg">
   Generalized Hardy-Weinberg
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#slim-simulations-with-migration">
     SLiM simulations with migration
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="migration-and-gene-flow">
<h1>Migration and Gene Flow<a class="headerlink" href="#migration-and-gene-flow" title="Permalink to this headline">¶</a></h1>
<p>In population genetics, the term “migration” is really meant to
describe Gene flow, defined as the movement of alleles from one area
(deme, population, region) to another. Gene flow assumes some form of
dispersal or migration (wind pollination, seed dispersal, birds flying,
etc.) but dispersal is not gene flow (genes must be transferred, not
just their carriers)</p>
<p>We are going to build a model of gene flow in exactly the same way we
studied mutation. Consider two populations, a mainland population and an
island population. Each of these populations has the <span class="math notranslate nohighlight">\(A_1\)</span> allele at
frequencies <span class="math notranslate nohighlight">\(p_{main}\)</span> and <span class="math notranslate nohighlight">\(p_{island}\)</span> respectively. Assume that gene
flow is one way, from mainland to island and that the proportion of
individuals who become parents in the island population is <span class="math notranslate nohighlight">\(m\)</span>. Although
I’ve said this is a proportion, also notice that we could consider this
a probability interchangeably. After a round of migration, in the island
population there are then two sources for alleles, they could be from
the island population originally with probability <span class="math notranslate nohighlight">\(1-m\)</span>, or they could
have migrated from the mainland with probability <span class="math notranslate nohighlight">\(m\)</span>. This means that
after migration the allele frequency of <span class="math notranslate nohighlight">\(A_1\)</span> in the island population
is
$<span class="math notranslate nohighlight">\(\begin{aligned}
p_{island}' = p_{island}(1-m) + p_{main}m.
\end{aligned}\)</span>$</p>
<p>Simple enough right? Now lets to what comes naturally and study the change in allele
frequency as a result of mutation. Following what we have done in our
other analyses,</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
\Delta_{m}p_{island}  &amp; = &amp; p_{island}' - p_{island} \\
 &amp; = &amp; p_{island}(1-m) + p_{main}m - p_{island} \\
 &amp; = &amp; m(p_{main} - p_{island}) \\
\end{aligned}\end{split}\]</div>
<p>Beautiful. Now we have
a very simple expression for how allele frequencies in the island
population should change due to gene flow from the mainland population.
This change in allele frequency makes sense- it only depends on the
amount of migration and the differences in allele frequencies between
the two populations. It’s simple enough to generalize this to multiple
populations, or to populations at different distances away from one
another.</p>
<p>Let’s plot <span class="math notranslate nohighlight">\(\Delta_{m}p_{island}\)</span></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> slim_magic

<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span><span class="p">,</span> <span class="n">SVG</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">delta_m</span><span class="p">(</span><span class="n">p_m</span><span class="p">,</span> <span class="n">p_i</span><span class="p">,</span> <span class="n">m</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">m</span> <span class="o">*</span> <span class="p">(</span><span class="n">p_m</span> <span class="o">-</span> <span class="n">p_i</span><span class="p">)</span>

<span class="n">x</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">100</span><span class="p">)</span>
<span class="n">p_m</span> <span class="o">=</span> <span class="mf">0.5</span>
<span class="n">p_i</span> <span class="o">=</span> <span class="mf">0.5</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">delta_m</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">p_m</span><span class="p">,</span> <span class="n">p_i</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">vlines</span><span class="p">(</span><span class="n">p_i</span><span class="p">,</span><span class="o">-</span><span class="mf">0.3</span><span class="p">,</span> <span class="mf">0.3</span><span class="p">,</span> <span class="n">linestyles</span><span class="o">=</span><span class="s1">&#39;dashed&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;$\Delta m$&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;$p_m$&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0.5, 0, &#39;$p_m$&#39;)
</pre></div>
</div>
<img alt="_images/lecture4_3_1.png" src="_images/lecture4_3_1.png" />
</div>
</div>
<p>the dotted line here represents the frequency of the island population. Ask yourself, why does
<span class="math notranslate nohighlight">\(\Delta_{m}p_{island} = 0\)</span> at that point?</p>
<div class="section" id="non-random-mating">
<h2>Non-random mating<a class="headerlink" href="#non-random-mating" title="Permalink to this headline">¶</a></h2>
<p>When individuals do not mate randomly, the assumptions of the Hardy-Weinberg
law are no longer met. A potent force that leads to non-random mating is population
subdivision, where for instance individuals in certain parts of the range
are more likely to mate with one another than they are with individuals that
are far away or in a separate patch connected via migration.</p>
<p>Let’s motivate this with an example simulation from SLiM, that’s a spin on
one we looked at on the first day.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_rstack</span> 10 --out subdivision
// set up a single locus simulation of drift
initialize()
{
    // set the overall mutation rate
    initializeMutationRate(0);
    // m1 mutation type: neutral
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, 0.0);
    // g1 genomic element type: uses m1 probability 1
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    // uniform chromosome of length 1 site
    initializeGenomicElement(g1, 0, 0);
    // uniform recombination along the chromosome
    initializeRecombinationRate(0);
    suppressWarnings(T);
}

// create 2 populations of 50 individuals
1 {
    sim.addSubpop(&quot;p1&quot;, 50);
    sim.addSubpop(&quot;p2&quot;, 50);
    // sample 50 haploid genomes from the first population
    target = sample(p1.genomes, 50);
    // add a mutation to those genomes
    target.addNewMutation(m1,0, 0);
    //print header for the output
    cat(&quot;generation,p,x11,x12,x22\\n&quot;);
}
1:300 late(){
    inds1 = p1.sampleIndividuals(50);
    inds2 = p2.sampleIndividuals(50);
    inds = c(inds1, inds2);
    ind_count = inds.countOfMutationsOfType(m1);
    counts = c(0, 0, 0);
    for (x in ind_count)
        counts[x] = counts[x] + 1;
    counts = counts / 100;
    freqs1 = sim.mutationFrequencies(p1);
    freqs2 = sim.mutationFrequencies(p2);
    freqs = freqs1 + freqs2;
    if (length(freqs) &gt; 0.0)
        catn(sim.generation + &quot;,&quot; + freqs + &quot;,&quot; + paste(counts, sep=&quot;,&quot;));
    }
// run to generation 0
300 late() {
    sim.simulationFinished();
    }
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">hwe</span><span class="p">(</span><span class="n">p</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">p</span><span class="o">**</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span> <span class="o">*</span> <span class="n">p</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">),</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">)</span><span class="o">**</span><span class="mi">2</span><span class="p">])</span>

<span class="c1">#run that function for 100 pts between (0,1)</span>
<span class="n">expected</span> <span class="o">=</span> <span class="n">hwe</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">))</span>

<span class="c1">#plot simulated</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">subdivision</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">subdivision</span><span class="o">.</span><span class="n">x11</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_11 frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">subdivision</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">subdivision</span><span class="o">.</span><span class="n">x12</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_12 frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">subdivision</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">subdivision</span><span class="o">.</span><span class="n">x22</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_22 frequency&quot;</span><span class="p">)</span>


<span class="c1">#plot expected</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">0</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;HWE expected&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">1</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">2</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;allele frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;genotype frequency&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0, 0.5, &#39;genotype frequency&#39;)
</pre></div>
</div>
<img alt="_images/lecture4_7_1.png" src="_images/lecture4_7_1.png" />
</div>
</div>
<p>So we can see in this example of two, subdivided populations which don’t share migrants that we
are deviating sharply from our Hardy Weinberg expectations. In this specific case you might notice
that there are <em>too few</em> heterozygotes relative to our expectations from Hardy Weinberg. Let’s
model this.</p>
</div>
<div class="section" id="generalized-hardy-weinberg">
<h2>Generalized Hardy-Weinberg<a class="headerlink" href="#generalized-hardy-weinberg" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>H-W Frequency:</p></td>
<td><p><span class="math notranslate nohighlight">\(p^2(1-F) + pF\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(2pq(1-F)\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(q^2(1-F) + qF\)</span></p></td>
</tr>
</tbody>
</table>
<p>We’ve introduced a new variable here, <span class="math notranslate nohighlight">\(F\)</span>, which the deviations from the expected frequencies.
If <span class="math notranslate nohighlight">\(F=0\)</span> we are back to our expected HW values. If <span class="math notranslate nohighlight">\(0 &lt; F \leq 1\)</span> then there is an excess of homozygotes
relative to expectations, if <span class="math notranslate nohighlight">\(F &lt; 0\)</span> there is an excess of heterozygotes.</p>
<p>Next lets look a bit more closely at a flavor of <span class="math notranslate nohighlight">\(F\)</span> that we use in the particular case of subdivision
that we are simulating above. Let’s call that <span class="math notranslate nohighlight">\(Fst\)</span>. Next imagine a species arranged as above
in two discrete subpopulations that we could call subpop1 and subpop2. Next let’s imagine that the
current allele and genotype frequencies in each of my subpops is as follows:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Subpop1 Frequency:</p></td>
<td><p>0.25</p></td>
<td><p>0.0625</p></td>
<td><p>0.375</p></td>
<td><p>0.5625</p></td>
</tr>
<tr class="row-odd"><td><p>Subpop2 Frequency:</p></td>
<td><p>0.75</p></td>
<td><p>0.5625</p></td>
<td><p>0.375</p></td>
<td><p>0.0625</p></td>
</tr>
<tr class="row-even"><td><p>Species-wide Frequency:</p></td>
<td><p>0.5</p></td>
<td><p>0.3125</p></td>
<td><p>0.375</p></td>
<td><p>0.3125</p></td>
</tr>
<tr class="row-odd"><td><p>HWE Frequency:</p></td>
<td><p>0.5</p></td>
<td><p>0.25</p></td>
<td><p>0.5</p></td>
<td><p>0.25</p></td>
</tr>
</tbody>
</table>
<p>here is species-wide frequency is simple the average of the two subpopulations, assuming
they are of equal size. Here, even though each subpopulation has HW expected proportions
of genotype frequencies, the species as a whole does not. In fact there are
too few heterozygotes relative to our expectations. Let’s measure this with <span class="math notranslate nohighlight">\(Fst\)</span>,
we say</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
2pq(1-F_{st}) &amp; = 0.375 \\
(1-F_{st})  &amp; = 0.375 / 0.5 \\
F_{st} &amp;= 0.25
\end{aligned}\end{split}\]</div>
<p>In general, populations that have insufficient migration between them to homogenize
allele frequencies across the landscape with show positive <span class="math notranslate nohighlight">\(F_{st}\)</span> like this in pooled
samples. This deficiency of heterozygotes in cryptically sampled subpopulations is
sometimes called the Wahlund effect.</p>
<div class="section" id="slim-simulations-with-migration">
<h3>SLiM simulations with migration<a class="headerlink" href="#slim-simulations-with-migration" title="Permalink to this headline">¶</a></h3>
<p>Okay let’s modify our simulations above to include migration between populations. Let’s start with little migration
and then increase it back in.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_rstack</span> 10 --out subdivision
// set up a single locus simulation of drift
initialize()
{
    // set the overall mutation rate
    initializeMutationRate(0);
    // m1 mutation type: neutral
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, 0.0);
    // g1 genomic element type: uses m1 probability 1
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    // uniform chromosome of length 1 site
    initializeGenomicElement(g1, 0, 0);
    // uniform recombination along the chromosome
    initializeRecombinationRate(0);
    suppressWarnings(T);
}

// create 2 populations of 50 individuals
1 {
    sim.addSubpop(&quot;p1&quot;, 50);
    sim.addSubpop(&quot;p2&quot;, 50);
    // add migration between the two populations
    p1.setMigrationRates(p2, 0.001);
    p2.setMigrationRates(p1, 0.001);

    // sample 50 haploid genomes from the first population
    target = sample(p1.genomes, 50);
    // add a mutation to those genomes
    target.addNewMutation(m1,0, 0);
    //print header for the output
    cat(&quot;generation,p,x11,x12,x22\\n&quot;);
}
1:300 late(){
    inds1 = p1.sampleIndividuals(50);
    inds2 = p2.sampleIndividuals(50);
    inds = c(inds1, inds2);
    ind_count = inds.countOfMutationsOfType(m1);
    counts = c(0, 0, 0);
    for (x in ind_count)
        counts[x] = counts[x] + 1;
    counts = counts / 100;
    freqs1 = sim.mutationFrequencies(p1);
    freqs2 = sim.mutationFrequencies(p2);
    freqs = (freqs1 + freqs2) / 2;
    if (length(freqs) &gt; 0.0)
        catn(sim.generation + &quot;,&quot; + freqs + &quot;,&quot; + paste(counts, sep=&quot;,&quot;));
    }
// run to generation 0
300 late() {
    sim.simulationFinished();
    }
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#plot simulated</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">subdivision</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">subdivision</span><span class="o">.</span><span class="n">x11</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_11 frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">subdivision</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">subdivision</span><span class="o">.</span><span class="n">x12</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_12 frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">subdivision</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">subdivision</span><span class="o">.</span><span class="n">x22</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_22 frequency&quot;</span><span class="p">)</span>


<span class="c1">#plot expected</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">0</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;HWE expected&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">1</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">2</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;allele frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;genotype frequency&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0, 0.5, &#39;genotype frequency&#39;)
</pre></div>
</div>
<img alt="_images/lecture4_12_1.png" src="_images/lecture4_12_1.png" />
</div>
</div>
<p>what do you notice here? what is the pattern?</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="lecture3.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Mutation</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="lecture5.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Molecular Evolution</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Andrew Kern and Peter Ralph<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>