
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Genetic drift &#8212; Bio 610-- Intro to Population Genetics</title>
    
  <link href="_static/css/theme.css" rel="stylesheet">
  <link href="_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Genetic diversity, with trees" href="genetic_diversity.html" />
    <link rel="prev" title="Molecular Evolution" href="lecture5.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/tree.jpeg" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Bio 610-- Intro to Population Genetics</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Welcome to Population Genetics
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Population Genetics Concepts
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="lecture1.html">
   Introduction to Population Genetics
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture2.html">
   Natural Selection
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture3.html">
   Mutation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture4.html">
   Migration and Gene Flow
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture5.html">
   Molecular Evolution
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Genetic drift
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_diversity.html">
   Genetic diversity, with trees
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Exercises
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="exercise1.html">
   Exercise 1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="exercise2.html">
   Exercise 2
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/genetic_drift.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/kr-colab/bio610_popgen"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/kr-colab/bio610_popgen/issues/new?title=Issue%20on%20page%20%2Fgenetic_drift.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/kr-colab/bio610_popgen/main?urlpath=lab/tree/book/genetic_drift.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Genetic drift
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-simplest-model">
     The simplest model
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#okay-now-the-theory">
     Okay, now the theory
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#exercise">
       Exercise:
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mutation-drift-equilibrium">
   Mutation-drift equilibrium
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#a-note-on-notation">
     A note on notation
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#theory-again">
     Theory, again
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fixation-rates">
   Fixation rates
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Genetic drift</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Genetic drift
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-simplest-model">
     The simplest model
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#okay-now-the-theory">
     Okay, now the theory
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#exercise">
       Exercise:
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mutation-drift-equilibrium">
   Mutation-drift equilibrium
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#a-note-on-notation">
     A note on notation
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#theory-again">
     Theory, again
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fixation-rates">
   Fixation rates
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="genetic-drift">
<h1>Genetic drift<a class="headerlink" href="#genetic-drift" title="Permalink to this headline">¶</a></h1>
<p>Natural selection can push allele frequencies around,
but what if it isn’t?
Allele frequencies still change, due to the randomness inherent in who reproduces
and who doesn’t.
Or, what if natural selection is acting – how important is the randomness?</p>
<div class="section" id="the-simplest-model">
<h2>The simplest model<a class="headerlink" href="#the-simplest-model" title="Permalink to this headline">¶</a></h2>
<p>As usual, let’s start simple.
Suppose in a population of size <span class="math notranslate nohighlight">\(N\)</span> an allele is at frequency <span class="math notranslate nohighlight">\(p = k/N\)</span>.
On average, we’d expect it to <em>still</em> be at frequency <span class="math notranslate nohighlight">\(p\)</span> in the next generation.
Let’s see how that works out.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> slim_magic

<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span><span class="p">,</span> <span class="n">SVG</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_cstack</span> 30 --out drift
// set up a single locus simulation of drift
initialize()
{
    initializeMutationRate(0);
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, 0.0);
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    initializeGenomicElement(g1, 0, 0);
    initializeRecombinationRate(0);
    suppressWarnings(T);
}

1 {
    sim.addSubpop(&quot;p1&quot;, 500);
    // add mutation to half the genomes from the first population
    target = sample(p1.genomes, p1.individualCount);
    target.addNewMutation(m1,0, 0);
    log = sim.createLogFile(&quot;/dev/stdout&quot;, logInterval=1);
    log.addGeneration();
    log.addCustomColumn(&quot;p&quot;, &quot;mean(p1.individuals.countOfMutationsOfType(m1))/2;&quot;);
}

500 late() {
    sim.simulationFinished();
}
</pre></div>
</div>
</div>
</div>
<p>Here’s those allele frequencies through time:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">drift</span><span class="o">.</span><span class="n">set_axis</span><span class="p">([</span><span class="sa">f</span><span class="s2">&quot;p</span><span class="si">{</span><span class="n">k</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">drift</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">])],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;number of generations&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;allele frequency&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">drift</span><span class="p">:</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">drift</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">drift</span><span class="p">[</span><span class="n">col</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/genetic_drift_5_0.png" src="_images/genetic_drift_5_0.png" />
</div>
</div>
<p>And, here’s the standard deviation of those allele frequences through time:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">p</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">drift</span><span class="p">)</span>
<span class="n">sds</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span>
    <span class="n">np</span><span class="o">.</span><span class="n">std</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">p</span>
<span class="p">])</span>

<span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">drift</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">sds</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;number of generations&quot;</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;SD of allele frequency&quot;</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">drift</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">sds</span><span class="o">**</span><span class="mi">2</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;number of generations&quot;</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;variance of allele frequency&quot;</span><span class="p">);</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/genetic_drift_7_0.png" src="_images/genetic_drift_7_0.png" />
</div>
</div>
</div>
<div class="section" id="okay-now-the-theory">
<h2>Okay, now the theory<a class="headerlink" href="#okay-now-the-theory" title="Permalink to this headline">¶</a></h2>
<p>Why’s that?
Well, think about how the WF model works:
each offspring genome is produced by a uniformly randomly selected parent genome
(independently of everything else).
So, if <span class="math notranslate nohighlight">\(X_t\)</span> is the number of copies of the allele we’re counting at generation <span class="math notranslate nohighlight">\(t\)</span>,
then</p>
<div class="math notranslate nohighlight">
\[
   X_{t+1} \sim \text{Binomial}(2N, X_t/2N) .
\]</div>
<p>In other words, we get the next generation “by binomial sampling”.
Since the allele frequency is <span class="math notranslate nohighlight">\(P_t = X_t/2N\)</span>,
then given this,</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
  \mathbb{E}[X_{t+1}] &amp;= X_t \\
  \mathbf{var}[X_{t+1}] &amp;= X_t (1 - X_t / 2N) ,
\end{aligned}\end{split}\]</div>
<p>and so</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
  \mathbb{E}[P_{t+1}] &amp;= P_t \\
  \mathbf{var}[P_{t+1}] &amp;= P_t (1 - P_t) / 2N .
\end{aligned}\end{split}\]</div>
<p>If we assume that the frequency doesn’t change too much over time,
then</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
  \mathbb{E}[P_{t}] &amp;= P_0 \\
  \mathbf{var}[P_{t}] &amp;\approx t P_0 (1 - P_0) / 2N .
\end{aligned}\end{split}\]</div>
<div class="section" id="exercise">
<h3>Exercise:<a class="headerlink" href="#exercise" title="Permalink to this headline">¶</a></h3>
<p>Put the “theory” lines on the plots above.</p>
</div>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="mutation-drift-equilibrium">
<h1>Mutation-drift equilibrium<a class="headerlink" href="#mutation-drift-equilibrium" title="Permalink to this headline">¶</a></h1>
<p>The next question is: mutation adds genetic variation and drift removes it;
where do these balance? We’ll look at this a few different ways.
First, a simulation:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_cstack</span> 3 --out het
// set up a single locus simulation of drift
initialize()
{
    initializeMutationRate(1e-8);
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, 0.0);
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    initializeGenomicElement(g1, 0, 9999999);
    initializeRecombinationRate(1e-8);
    suppressWarnings(T);
}

1 {
    sim.addSubpop(&quot;p1&quot;, 500);
    // add mutation to half the genomes from the first population
    target = sample(p1.genomes, p1.individualCount);
    target.addNewMutation(m1,0, 0);
    log = sim.createLogFile(&quot;/dev/stdout&quot;, logInterval=1);
    log.addGeneration();
    log.addCustomColumn(
        &quot;pi&quot;,
        &quot;calcHeterozygosity(p1.individuals.genomes);&quot;
    );
}

5000 late() {
    sim.simulationFinished();
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">het</span><span class="o">.</span><span class="n">set_axis</span><span class="p">([</span><span class="sa">f</span><span class="s2">&quot;pi</span><span class="si">{</span><span class="n">k</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">het</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">])],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;number of generations&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;heterozygosity&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">het</span><span class="p">:</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">het</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">het</span><span class="p">[</span><span class="n">col</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/genetic_drift_13_0.png" src="_images/genetic_drift_13_0.png" />
</div>
</div>
<p>To understand what that’s showing us, here’s what the SLiM function <code class="docutils literal notranslate"><span class="pre">calcHeterozygosity(</span> <span class="pre">)</span></code> does:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>	<span class="n">length</span> <span class="o">=</span> <span class="n">sim</span><span class="o">.</span><span class="n">chromosome</span><span class="o">.</span><span class="n">lastPosition</span> <span class="o">+</span> <span class="mi">1</span><span class="p">;</span>
	<span class="n">p</span> <span class="o">=</span> <span class="n">genomes</span><span class="o">.</span><span class="n">mutationFrequenciesInGenomes</span><span class="p">(</span><span class="n">muts</span><span class="p">);</span>
	<span class="n">heterozygosity</span> <span class="o">=</span> <span class="mi">2</span> <span class="o">*</span> <span class="nb">sum</span><span class="p">(</span><span class="n">p</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">))</span> <span class="o">/</span> <span class="n">length</span><span class="p">;</span>
	<span class="k">return</span> <span class="n">heterozygosity</span><span class="p">;</span>
</pre></div>
</div>
<p>This is the expected proportion of positions
at which two randomly chosen genomes differ.
In this simulation, that number equilibrates around <span class="math notranslate nohighlight">\(2 \times 10^{-5}\)</span>.
What do we expect?</p>
<div class="section" id="a-note-on-notation">
<h2>A note on notation<a class="headerlink" href="#a-note-on-notation" title="Permalink to this headline">¶</a></h2>
<p>Above we’ve used “heterozygosity” to refer to a <em>statistic</em> calculated
from a set of genomes, that summarizes the amount of genetic diversity within those genomes.
This is sometimes called <span class="math notranslate nohighlight">\(\pi\)</span>.
Population genetics literature is not known for having consistent notation,
or defining its terms,
and indeed, <span class="math notranslate nohighlight">\(\pi\)</span> is also often used to refer to the <em>expected value</em> of that statistic
under some model or other.
This confuses the actual value of the statistic calculated from a dataset
with its mean under some model;
actually, the value of the statistic computed from data produced by that model
will be random, and vary about the expected value
to some extent.</p>
<p>We can see this in the plots above:
there is noisiness in the trajectories,
and the replicates don’t converge to the same value,
although they do wander around some value.
Our goal next is to understand this equilibrium value -
the long-term mean.
We’ll refer to the statistic one would compute from the data
with a Roman letter, <span class="math notranslate nohighlight">\(h\)</span>,
and the expected value under a model with a greek one: <span class="math notranslate nohighlight">\(\mathbb{E}[h] = \pi\)</span>.</p>
</div>
<div class="section" id="theory-again">
<h2>Theory, again<a class="headerlink" href="#theory-again" title="Permalink to this headline">¶</a></h2>
<p>Consider that definition of (expected) heterozygosity -
the expected number of positions at which two randomly chosen genomes differ.
When do two genomes differ at a given position?
Well, they differ either if their parents differ,
or if they don’t and there’s been a mutation while producing one of their genomes.
The  chance the two parents differ is just the heterozygosity in the previous generation,
except for the case in which the two genomes actually share a common parent.
We can use this basic observation to write a recursion for <span class="math notranslate nohighlight">\(\pi_t\)</span>,
the expected heterozygosity at time <span class="math notranslate nohighlight">\(t\)</span>.</p>
<p>Let’s take a simple model of mutation - at each site there are two alleles,
and each mutates to the other with probability <span class="math notranslate nohighlight">\(\mu\)</span>.
Then, the probability that siblings differ at a site is <span class="math notranslate nohighlight">\(2 \mu (1-\mu)\)</span>.
Since the probability under the Wright-Fisher model that the two genomes are siblings
at a given site is <span class="math notranslate nohighlight">\(1/2N\)</span>,
the previous paragraph says that</p>
<div class="math notranslate nohighlight">
\[
    \pi_{t+1} = 
        \frac{1}{2N} 2 \mu (1-\mu)
        + \left(1 - \frac{1}{2N} \right)
          \left\{
              \pi_t (1 - 2 \mu (1-\mu))
              + (1 - \pi_t) 2 \mu (1-\mu)
          \right\} .
\]</div>
<p>Here the terms are:
share parents and mutation;
parents differ and no mutation;
parents do not differ and mutation.
Rearranging, this is</p>
<div class="math notranslate nohighlight">
\[
    \pi_{t+1} = 
        2 \mu (1-\mu)
        + \left(1 - \frac{1}{2N} \right)
              (1 - 4 \mu (1-\mu))
              \pi_t .
\]</div>
<p>To solve for stationarity,
let’s ask what value of <span class="math notranslate nohighlight">\(\pi_*\)</span> would have <span class="math notranslate nohighlight">\(\pi_t = \pi_{t+1} = \pi_*\)</span>.
Solving for <span class="math notranslate nohighlight">\(\pi_*\)</span>, we get</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
    \pi_* &amp;=
    \frac{
        4 N \mu (1 - \mu)
    }{
        1 + 4 \mu (1 - \mu)(2 N - 1)
    } \\
    &amp;\approx 4 N \mu .
\end{aligned}\end{split}\]</div>
<p><strong>Exercise:</strong>
Add the theory curves to the plot above.</p>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="fixation-rates">
<h1>Fixation rates<a class="headerlink" href="#fixation-rates" title="Permalink to this headline">¶</a></h1>
<p>The next aspect of drift is:
what proportion of new mutations end up fixing in the population?
For neutral mutations, the answer is easy:
imagine the population at the time when a new, neutral mutation occurs
at a given position.
That neutral mutation has equal chance of occurring in any of the <span class="math notranslate nohighlight">\(2N\)</span> genomes.
Now imagine in the far future - eventually, everyone in the future will
descend from just a single one of these genomes.
The chance that the mutation picks that one lucky “ultimate ancestor”
is just <span class="math notranslate nohighlight">\(1/2N\)</span>.</p>
<p>So, if we write <span class="math notranslate nohighlight">\(p_f(s)\)</span> for the probability of fixation
of a new mutation with selection coefficient <span class="math notranslate nohighlight">\(s\)</span>,
then in a population of diploid size <span class="math notranslate nohighlight">\(N\)</span>,</p>
<div class="math notranslate nohighlight">
\[
  p_f(0) = \frac{1}{2N} .
\]</div>
<p><strong>Exercise:</strong> Check this, with a simulation.</p>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="lecture5.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Molecular Evolution</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="genetic_diversity.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Genetic diversity, with trees</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Andrew Kern and Peter Ralph<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>