
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Mutation &#8212; Bio 610-- Intro to Population Genetics</title>
    
  <link href="_static/css/theme.css" rel="stylesheet">
  <link href="_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Migration and Gene Flow" href="lecture4.html" />
    <link rel="prev" title="Natural Selection" href="lecture2.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/tree.jpeg" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Bio 610-- Intro to Population Genetics</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Welcome to Population Genetics
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Population Genetics Concepts
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="lecture1.html">
   Introduction to Population Genetics
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture2.html">
   Natural Selection
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Mutation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture4.html">
   Migration and Gene Flow
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture5.html">
   Molecular Evolution
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_drift.html">
   Genetic drift
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_diversity.html">
   Genetic diversity, with trees
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Exercises
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="exercise1.html">
   Exercise 1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="exercise2.html">
   Exercise 2
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/lecture3.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/kr-colab/bio610_popgen"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/kr-colab/bio610_popgen/issues/new?title=Issue%20on%20page%20%2Flecture3.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/kr-colab/bio610_popgen/main?urlpath=lab/tree/book/lecture3.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Mutation
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mutation-and-evolution">
     Mutation and Evolution
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mutation-selection-balance">
   Mutation-Selection Balance
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Mutation</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Mutation
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mutation-and-evolution">
     Mutation and Evolution
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mutation-selection-balance">
   Mutation-Selection Balance
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="mutation">
<h1>Mutation<a class="headerlink" href="#mutation" title="Permalink to this headline">¶</a></h1>
<div class="section" id="mutation-and-evolution">
<h2>Mutation and Evolution<a class="headerlink" href="#mutation-and-evolution" title="Permalink to this headline">¶</a></h2>
<p>Previously we learned about how selection can change the frequencies of
alleles and genotypes in populations. Selection typically eliminates
variation from within populations. (The general exception to this claim
is with the class selection models we have called “balancing” selection
where alleles are maintained in the population by overdominance,
habitat-specific selection, or frequency dependent selection). If
selection removes variation, soon there will be no more variation for
selection to act on, and evolution will grind to a halt, right? This
might be true if it were not for the reality of mutation which will
restore genetic variation eliminated by selection. Thus, mutations are
the fundamental raw material of evolution.</p>
<p>The basic model of mutation that we will study is one way mutation. This
is when one allele through mutation can turn in to another such that</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
A_1 \stackrel{u}\longrightarrow A_2
\end{aligned}\]</div>
<p><span class="math notranslate nohighlight">\(u\)</span> here represents the mutation rate, the probability that a mutation
from <span class="math notranslate nohighlight">\(A_1\)</span> to <span class="math notranslate nohighlight">\(A_2\)</span> occurs during a meiosis.</p>
<p>It should be obvious that mutation will change allele frequencies. This
is true because there is a constant flux from <span class="math notranslate nohighlight">\(A_1\)</span> to <span class="math notranslate nohighlight">\(A_2\)</span> purely as a
result of this mutation process. We can study the change in allele
frequency due to mutation in a very similar way to how we studied the
change in allele frequency due to selection. Consider a population with
frequency of the <span class="math notranslate nohighlight">\(A_1\)</span> allele <span class="math notranslate nohighlight">\(p\)</span>. In the next generation, after a round
of mutation, each <span class="math notranslate nohighlight">\(A_1\)</span> allele must have been <span class="math notranslate nohighlight">\(A_1\)</span> in the current
generation and it must not have mutated. That is</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
    p' = p (1-u).
\end{aligned}\]</div>
<p>Now lets turn our attention to the
change in allele frequency in one generation due to mutation as we did
previously for selection</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
    \Delta_up &amp; = &amp; p' - p \\
     &amp; = &amp; p(1-u) -p \\
     &amp; = &amp; -up \\
\end{aligned}\end{split}\]</div>
<p>notice that our notation– <span class="math notranslate nohighlight">\(\Delta_up\)</span>
– emphasizes the source of the change in allele frequency is mutation.
Let’s plot this as a function of <span class="math notranslate nohighlight">\(p\)</span></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="k">def</span> <span class="nf">delta_p_u</span><span class="p">(</span><span class="n">u</span><span class="p">,</span> <span class="n">p</span><span class="p">):</span>
    <span class="k">return</span> <span class="o">-</span><span class="n">u</span> <span class="o">*</span> <span class="n">p</span>

<span class="n">x</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">100</span><span class="p">)</span>
<span class="n">frequencies</span> <span class="o">=</span> <span class="n">delta_p_u</span><span class="p">(</span><span class="mf">0.001</span><span class="p">,</span> <span class="n">x</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">frequencies</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[&lt;matplotlib.lines.Line2D at 0x7f85981df3a0&gt;]
</pre></div>
</div>
<img alt="_images/lecture3_2_1.png" src="_images/lecture3_2_1.png" />
</div>
</div>
<p>right! <em>decreasing</em> for all <span class="math notranslate nohighlight">\(p\)</span> except the case where <span class="math notranslate nohighlight">\(p = 0\)</span>!</p>
<hr class="docutils" />
<p><strong>Exercise</strong> write a function to explore the dynamics of allele frequency dynamics of this model over time. It should take an initial frequency and a mutation rate as it’s input, and output
the allele frequency in subsequent generations (perhaps the number of generations could be an input)</p>
<hr class="docutils" />
<p>This sort of unidirectional mutation acts to consistently decrease the
frequency of the <span class="math notranslate nohighlight">\(A_1\)</span> allele from generation to generation. If we
instead were to study two way mutation, the mutational flux would depend
on the proportional rates to and from <span class="math notranslate nohighlight">\(A_1\)</span>. So mutation, although it is
a random process with respect to target, leads to deterministic effects
on allele frequencies. Neat huh?</p>
<p>Mutation rates per generation are very very small. You know this
intuitively- think about cloning plants from cuttings. In <em>Drosophila</em>,
which is one of the best studied animals from the perspective of rates
of spontaneous mutation, the mutation rate per generation per nucleotide
is on the order of <span class="math notranslate nohighlight">\(10^{-9}\)</span>. This means that mutation changes the
frequency of alleles at a very slow rate. If we assume that there is
sufficiently strong selection against the <span class="math notranslate nohighlight">\(A_2\)</span> allele (i.e.
<span class="math notranslate nohighlight">\(w_{11} &gt;&gt; w_{22}\)</span>), then we can further approximate our change in
allele frequency due to mutation</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
\Delta_up  &amp; = &amp; -up \\
  &amp; = &amp; -u + qu \\
 &amp; \approx &amp; -u, 
\end{aligned}\end{split}\]</div>
<p>because <span class="math notranslate nohighlight">\(q \approx 0\)</span>. So in the case
of a deleterious <span class="math notranslate nohighlight">\(A_2\)</span> allele, we can see that the change in allele
frequency due to mutation is independent of allele frequencies. This is
our first hint that mutation and selection might combine in interesting
and important ways.</p>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="mutation-selection-balance">
<h1>Mutation-Selection Balance<a class="headerlink" href="#mutation-selection-balance" title="Permalink to this headline">¶</a></h1>
<p>We can imagine mutation and selection as opposing forces which might
come to some equilibrium in terms of the number or frequency of
deleterious alleles within a population. To make this concrete think
about the human genetic disease cystic fibrosis (CF). CF is a very
serious genetic disorder in which a transmembrane protein in lung
epithelium cells called CFTR is non-functional. Hundreds, if not
thousands of separate mutations in CFTR lead to CF, thus we could
imagine that there is a certain, appreciable rate of mutation to CF. If
each of these mutations is deleterious (i.e. they cause disease) then
over generations they should be selected out of the population. Thus
mutation will inject CF mutations into the population, but selection
will remove them- can we study this as an equilibrium process?</p>
<p>Our approach will be to study each of our evolutionary forces in
isolation, and then combine them to figure out how they interact. Let’s
start by considering the change in allele frequency due to selection
that we studied in lecture 7, but this time we will approximate it under
the assumption that <span class="math notranslate nohighlight">\(q \approx 0\)</span></p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
\Delta_sp  &amp; = &amp;\frac{pqs[ph + q(1-h)]}{\bar{w}} \\
&amp; \approx &amp; qhs 
\end{aligned}\end{split}\]</div>
<p>This approximation goes down the road
because when <span class="math notranslate nohighlight">\(q \approx 0\)</span>, <span class="math notranslate nohighlight">\(p \approx 1\)</span>, <span class="math notranslate nohighlight">\(\bar{w} \approx 1\)</span>, and we
can ignore all terms of order <span class="math notranslate nohighlight">\(q^2\)</span>.</p>
<p>Now lets combine the forces of selection and mutation on the change in
frequency of <span class="math notranslate nohighlight">\(A_1\)</span> using the approximations we have just derived
(equations 3 and 5). At equilibrium that change in allele frequency due
to the combined actions of mutation and selection must equal zero. That
is</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{aligned}
    0 &amp; = &amp; \Delta_up + \Delta_sp \\
    &amp; \approx &amp; -u + qhs
\end{aligned}\end{split}\]</div>
<p>so the equilibrium frequency of
the <span class="math notranslate nohighlight">\(A_2\)</span> is</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
    \hat{q} \approx \frac{u}{hs}
\end{aligned}\]</div>
<p>Thus we see that
deleterious (e.g. disease) allele frequencies are determined by both the
mutation rate to those alleles and their selective effects in
heterozygotes. As we saw earlier, new mutations overwhelmingly are found
in heterozygous states, so it’s perhaps not surprising that <span class="math notranslate nohighlight">\(h\)</span> should
dominate the fate of deleterious alleles.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> slim_magic

<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span><span class="p">,</span> <span class="n">SVG</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_cstack</span> 100 --out neutral
initialize()
{
    // set the overall mutation rate
    // no mutation in this simulation
    initializeMutationRate(0.01);
    // m1 mutation type: neutral
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, 0.0);
    // single locus simulation
    // g1 genomic element type: uses m1 probability 1
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    // uniform chromosome of length 1 site
    initializeGenomicElement(g1, 0, 0);
    // uniform recombination along the chromosome
    initializeRecombinationRate(0);
    suppressWarnings(T);
}

// create a population of 100 individuals
1 {
    sim.addSubpop(&quot;p1&quot;, 100);
    // sample 100 haploid genomes 
    target = sample(p1.genomes, 100);
    // add a mutation to those genomes
    // H_0 = 0.5 here
    target.addNewMutation(m1,0, 0);
    cat(&quot;generation,p\\n&quot;);
}
1:1000 late(){
    // allele freq of A_1 allele in current gen
    freqs = sim.mutationFrequencies(p1);
    // each gen print out the gen and the allele freqs 
    catn(sim.generation + &quot;,&quot; + freqs[0]);
    }
// run to generation 0
300 late() {
    sim.simulationFinished();
    }
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_cstack</span> 100 --out selection
// set up a single locus simulation of selection
initialize()
{
    // set the overall mutation rate
    // no mutation in this simulation
    initializeMutationRate(0.01);
    // m1 mutation type: neutral
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, -0.1);
    // single locus simulation
    // g1 genomic element type: uses m1 probability 1
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    // uniform chromosome of length 1 site
    initializeGenomicElement(g1, 0, 0);
    // uniform recombination along the chromosome
    initializeRecombinationRate(0);
    suppressWarnings(T);
}

// create a population of 100 individuals
1 {
    sim.addSubpop(&quot;p1&quot;, 100);
    cat(&quot;generation,p\\n&quot;);
}
1:1000 late(){
    // allele freq of A_1 allele in current gen
    freqs = sim.mutationFrequencies(p1);
    // each gen print out the gen and the allele freqs 
    if( length(freqs) &gt; 0)
        catn(sim.generation + &quot;,&quot; + freqs[0]);
    }
// run to generation 0
300 late() {
    sim.simulationFinished();
    }
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plt</span><span class="o">.</span><span class="n">hist</span><span class="p">(</span><span class="n">neutral</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span> <span class="n">density</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;neutral&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">hist</span><span class="p">(</span><span class="n">selection</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span> <span class="n">density</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;selection&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/lecture3_10_0.png" src="_images/lecture3_10_0.png" />
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="lecture2.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Natural Selection</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="lecture4.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Migration and Gene Flow</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Andrew Kern and Peter Ralph<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>