
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Introduction to Population Genetics &#8212; Bio 610-- Intro to Population Genetics</title>
    
  <link href="_static/css/theme.css" rel="stylesheet">
  <link href="_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Natural Selection" href="lecture2.html" />
    <link rel="prev" title="Welcome to Population Genetics" href="intro.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/tree.jpeg" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Bio 610-- Intro to Population Genetics</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Welcome to Population Genetics
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Population Genetics Concepts
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Introduction to Population Genetics
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture2.html">
   Natural Selection
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture3.html">
   Mutation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="lecture4.html">
   Migration and Gene Flow
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_drift.html">
   Genetic drift
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="genetic_diversity.html">
   Genetic diversity, with trees
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="molecular_evolution.html">
   Molecular Evolution
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Exercises
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="exercise1.html">
   Exercise 1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="exercise2.html">
   Exercise 2
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/lecture1.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/kr-colab/bio610_popgen"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/kr-colab/bio610_popgen/issues/new?title=Issue%20on%20page%20%2Flecture1.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/kr-colab/bio610_popgen/main?urlpath=lab/tree/book/lecture1.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Introduction to Population Genetics
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-population-genetics">
   What is Population Genetics?
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#dna-variation">
   DNA variation
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#adh-in-drosophila-a-case-study">
     <em>
      Adh
     </em>
     in Drosophila: a case study
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#loci-alleles-and-identity">
   Loci, Alleles, and Identity
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genotype-and-allele-frequencies">
   Genotype and Allele Frequencies
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-hardy-weinberg-law">
   The Hardy-Weinberg Law
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#heterozygosity">
   Heterozygosity
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#simulating-hardy-weinberg-expectations">
     Simulating Hardy-Weinberg Expectations
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Introduction to Population Genetics</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Introduction to Population Genetics
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-population-genetics">
   What is Population Genetics?
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#dna-variation">
   DNA variation
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#adh-in-drosophila-a-case-study">
     <em>
      Adh
     </em>
     in Drosophila: a case study
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#loci-alleles-and-identity">
   Loci, Alleles, and Identity
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genotype-and-allele-frequencies">
   Genotype and Allele Frequencies
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-hardy-weinberg-law">
   The Hardy-Weinberg Law
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#heterozygosity">
   Heterozygosity
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#simulating-hardy-weinberg-expectations">
     Simulating Hardy-Weinberg Expectations
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="introduction-to-population-genetics">
<h1>Introduction to Population Genetics<a class="headerlink" href="#introduction-to-population-genetics" title="Permalink to this headline">¶</a></h1>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="what-is-population-genetics">
<h1>What is Population Genetics?<a class="headerlink" href="#what-is-population-genetics" title="Permalink to this headline">¶</a></h1>
<p>Population genetics is primarily concerned with the genetic basis of
evolution. Through a quantitative description of the dynamics of genetic
variation within and between populations, population genetics has
established both a descriptive and predictive understanding of how DNA,
proteins, and genomes should differ among individuals. This leads to the
<em>ultimate</em> definition of evolution- the change in the genetic makeup
(allele and genotype frequencies) of a population over time.</p>
<p>Population genetics is distinct from nearly every other discipline in
biology in that important results are often theoretical, rather than
observational or experimental. The reason for this is that the object of
study, changes in genotype frequencies and fitnesses over time, are very
hard to observe in the lifetime of a human. If the relevant time frame
for evolutionary change is on the order of hundreds of thousands or
millions of years, than directly observing changes in genotypic
frequencies is impossible. Instead, population geneticists rely on
elegant mathematical models, first pioneered in the early part of the
20th century by Ronald Fisher, Sewall Wright, and J. B. S. Haldane, that
describe the changes of genotype frequencies as a result of evolutionary
forces. Progress is made by constructing population genetic models,
examining their behavior, and looking to see if the state of populations
is consistent with this behavior.</p>
<p>Fundamentally, genetic variation within populations is shaped by the
combined actions of predictable, <strong>deterministic</strong> evolutionary forces
and random, <strong>stochastic</strong> forces. The deterministic forces are
sometimes referred to as “linear pressures” in that they tend to push
allele frequencies in one direction (towards one, towards zero, or
towards an intermediate frequency). Important forces of this nature are
selection, mutation, gene flow, meiotic drive (unequal transmission of
certain alleles [a form of selection]), nonrandom mating (also a form
of selection). The primary stochastic evolutionary force is genetic
drift (potentially) which is due to the random sampling of individuals
(and genes) in finite populations. It is important to realize that the
deterministic forces may act together or against one another (e.g.,
selection may “try” to eliminate an allele that is pushed into the
population by recurrent mutation). Moreover, deterministic forces may
act with or against genetic drift, to determine the frequencies of
alleles and genotypes in populations (e.g., gene flow tends to
homogenize different populations while drift tends to make them
different). Hence, the interaction of these forces is what we are really
interested in (a later lecture), but since this can get very complex
mathematically, we will start by analyzing one force at a time.</p>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="dna-variation">
<h1>DNA variation<a class="headerlink" href="#dna-variation" title="Permalink to this headline">¶</a></h1>
<p>The fundamental unit of population genetic variation is a single DNA difference between individuals of a species. These differences might take the shape
of a so-called Simple Nucleotide Polymorphism (SNP) wherein an individual basepair has been swapped for another, or it could be that multiple basepairs have
been deleted in one individual relative to another. To make this all concrete, let’s focus on a single gene with a long history of study.</p>
<div class="section" id="adh-in-drosophila-a-case-study">
<h2><em>Adh</em> in Drosophila: a case study<a class="headerlink" href="#adh-in-drosophila-a-case-study" title="Permalink to this headline">¶</a></h2>
<p>The alcohol dehydrogenase gene, (<em>Adh</em>), is an incredibly well studied enzyme, both from the perspective of its biochemical properties, but also
as an early example of protein and genetic variation that was studied but many, many population geneticists. In the next few sections we’ll come
back to this locus again and again.</p>
<p>To orient us, I’ve taken Figure 1 directly from the now classic study DNA variation at <em>Adh</em> by
<a class="reference external" href="https://doi.org/10.1038/304412a0">Marty Kreitman (1983)</a> shows the structure of the <em>Adh</em> protein coding gene.
This is a pretty average gene in the drosophila genome– a few exons, separated by introns that are a few hundreds of bp long.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:center head"><p><img alt="Adh structure" src="_images/adh_structure_small.png" /></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:center"><p><em>Fig. 1.</em> Drosophila <em>Adh</em> gene structure, from Kreitman (1983)</p></td>
</tr>
</tbody>
</table>
<p>Kreitman, in what became a landmark study, described nucleotide variation in 11 strains of <em>Drosophila melanogaster</em> collected
from a worldwide sample. In what more or less corresponds to all the “raw” data, take a look at the table of nucleotide
variation shown in Figure 2. Here we see how sites vary according to their location in the gene structure.
There are a few things to note:</p>
<ul class="simple">
<li><p>there are variable positions in each part of the gene</p></li>
<li><p>there are <em>more</em> variable positions in introns than in exons on a percentage basis</p></li>
<li><p>some alleles (variant states) seem to be inherited <em>together</em> (more on linkage later)</p></li>
</ul>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:center head"><p><img alt="Adh polymorphism" src="_images/adh_poly_small.png" /></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:center"><p><em>Fig. 2.</em> DNA polymorphism at <em>Adh</em>, from Kreitman (1983)</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="loci-alleles-and-identity">
<h1>Loci, Alleles, and Identity<a class="headerlink" href="#loci-alleles-and-identity" title="Permalink to this headline">¶</a></h1>
<p>Before we go any further we need to get some nomenclature under our belts. The first distinction is that between <em>locus</em> and <em>alleles</em>.
A <strong>locus</strong> is the chromosomal address where a DNA sequence lives. An <strong>allele</strong> is the version of that DNA sequence found at a locus.
So we can talk about the <em>Adh</em> locus, the gene region on the chromosome, and we can talk about the <em>alleles</em> that we find at that locus
(e.g., those shown in Figure 2). Most organisms are diploid, that is they contain two complete copies of the genome, one inherited from mom
the other from dad. This means that any diploid individual may have two alleles at a locus, the maternal and paternal alleles.</p>
<p>Population genetics is concerned with describing patterns of allelic variation in natural populations of organism. Indeed for the population
geneticist there are three ways that alleles might differ from each other:</p>
<ul>
<li><p><strong>By origin</strong> alleles may differ by origin in a generation if they come from the same locus on different chromosomes in the previous
generation. For instance the alleles at a particular locus in a diploid individual differ by origin, because each was inherited from a
distinct parental chromosome (mom and dad). Also if we consider the 11 <em>Adh</em> alleles shown above, those are different by origin.</p>
<br>
</li>
<li><p><strong>By state</strong> alleles are said to be different by state if they, in someway, are different at the molecular level. For instance if we are
looking at DNA sequences, two alleles are different by state if they have different sequences. These alleles could differ at a single
nucleotide site or many, but either way they are said to differ by state. Moreover, we might not be considering DNA sequences at all.
Alleles can vary by state if they are different lengths, or have different electrophoretic mobilities on a gel, or could be in reference to
a specific position along a nucleotide or Amino Acid position. For instance, in Figure 2, there are (at least) two classes of alleles
that differ by state, the <em>Adh-f</em> and <em>Adh-s</em> alleles which correspond to the fast vs slow allozyme polymorphism.</p>
<br>
</li>
<li><p><strong>By descent</strong> allele differ by descent if they don’t share a common ancestor recently. In general all alleles at a locus are eventually
descended from a common ancestor, so this definition is really only useful in thinking about shorter timescales, say fewer than 10 generations.
Two alleles that differ by descent may or may not also differ by state.</p></li>
</ul>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="genotype-and-allele-frequencies">
<h1>Genotype and Allele Frequencies<a class="headerlink" href="#genotype-and-allele-frequencies" title="Permalink to this headline">¶</a></h1>
<p>To begin we need to understand some simple population genetic
“bookkeeping.” Consider a locus with two alleles (alternative forms of
the DNA sequence that “reside” at that locus, e.g., one from mother
other from father). Now consider a population of N individuals
(N=population size); this means that there are 2N alleles in the
population. We can thus talk about genotype frequencies and allele
frequencies. Let’s turn our attention to a hypothetical locus, the <span class="math notranslate nohighlight">\(A\)</span>
locus. We will first assume that there are two alleles segregating at
the <span class="math notranslate nohighlight">\(A\)</span> locus, <span class="math notranslate nohighlight">\(A_1\)</span> and <span class="math notranslate nohighlight">\(A_2\)</span>. These could be two allozyme alleles
(i.e. fast vs. slow) or two variant nucleotides at a polymorphic
position in a genome (i.e. a SNP). Three genotypes are possible at the
<span class="math notranslate nohighlight">\(A\)</span> locus: two homozygous genotypes, <span class="math notranslate nohighlight">\(A_1A_1\)</span> and <span class="math notranslate nohighlight">\(A_2A_2\)</span>, and one
heterozygous genotype <span class="math notranslate nohighlight">\(A_1A_2\)</span>. We write the relative frequency of a
genotype as <span class="math notranslate nohighlight">\(x_{ij}\)</span> as this table shows:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Relative Frequency:</p></td>
<td><p><span class="math notranslate nohighlight">\(x_{11}\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(x_{12}\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(x_{22}\)</span></p></td>
</tr>
</tbody>
</table>
<p>As relative frequencies must add to one we have $<span class="math notranslate nohighlight">\(\begin{aligned}
    x_{11} + x_{12} + x_{22} = 1 \end{aligned}\)</span>$</p>
<p>The subscripting of the heterozygotes is of course arbitrary. Following
my old advisor John Gillespie, we will take the convention of referring
to the heterozygote as <span class="math notranslate nohighlight">\(x_{12}\)</span> rather than <span class="math notranslate nohighlight">\(x_{21}\)</span> – you can only use
one subscripting convention if you want to keep things straight.</p>
<p>Often in population genetics we are interested in allele frequencies
rather than genotype frequencies. The frequency of the <span class="math notranslate nohighlight">\(A_1\)</span> allele in a
population is simply</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
    p = x_{11} + \frac{1}{2}x_{12},
\end{aligned}\]</div>
<p>and the frequency of the <span class="math notranslate nohighlight">\(A_2\)</span> allele is</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
    q = 1 - p =  x_{22} + \frac{1}{2}x_{12}.\end{aligned}\]</div>
<p>There are two ways to think about <span class="math notranslate nohighlight">\(p\)</span>, the frequency of the <span class="math notranslate nohighlight">\(A_1\)</span>
allele. The first way is as the relative frequency of the <span class="math notranslate nohighlight">\(A_1\)</span> allele
among all of the (two) alleles at the <span class="math notranslate nohighlight">\(A\)</span> locus in our population. This
is simple enough to get one’s head around. The second way to think about
<span class="math notranslate nohighlight">\(p\)</span> is as the probability that an allele picked by random from the
population is the <span class="math notranslate nohighlight">\(A_1\)</span> allele. Picking an allele at random is actually
composed of two parts, 1) picking a genotype at random from the
population and 2) then picking an allele from that genotype at random.
With our three genotypes at the <span class="math notranslate nohighlight">\(A\)</span> locus we could write down <span class="math notranslate nohighlight">\(p\)</span> as</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
    p =  (x_{11}\times1) + (x_{12}\times\frac{1}{2}) + (x_{22}\times0).\end{aligned}\]</div>
<p>This expression for <span class="math notranslate nohighlight">\(p\)</span> is broken down into three terms which reveal the
three mutually exclusive ways to sample an <span class="math notranslate nohighlight">\(A_1\)</span> allele from the
population and their associated probabilities. Let’s consider the first
term in this sum. This represents the joint event where we sample an
<span class="math notranslate nohighlight">\(A_1A_1\)</span> individual from our population (this occurs with probability
<span class="math notranslate nohighlight">\(x_{11}\)</span>) and then sample an <span class="math notranslate nohighlight">\(A_1\)</span> allele from that individual (this
occurs with probability one). In the second term we consider the joint
event of sampling an <span class="math notranslate nohighlight">\(A_1A_2\)</span> individual (with probability <span class="math notranslate nohighlight">\(x_{12}\)</span>) and
then sampling an <span class="math notranslate nohighlight">\(A_1\)</span> allele (with probability <span class="math notranslate nohighlight">\(\frac{1}{2}\)</span>). Now ask
yourself, what is that zero doing in the third term?? Got it?</p>
<p>Let’s use a concrete example to drive this all home. Consider a
population of <span class="math notranslate nohighlight">\(N = 100\)</span> individuals. We observe the following number of
genotypes at our <span class="math notranslate nohighlight">\(A\)</span> locus:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Observed Numbers:</p></td>
<td><p>25</p></td>
<td><p>50</p></td>
<td><p>25</p></td>
</tr>
<tr class="row-odd"><td><p>Relative Frequencies:</p></td>
<td><p>0.25</p></td>
<td><p>0.50</p></td>
<td><p>0.25</p></td>
</tr>
</tbody>
</table>
<p>The relative genotypic frequencies, the <span class="math notranslate nohighlight">\(x_{ij}s\)</span>, are simple to
calculate, just divide the number of each observed genotype by the
population size. For example, <span class="math notranslate nohighlight">\(x_{12} = \frac{50}{100} = 0.5\)</span>. Now let’s
calculate allele frequencies. The allele frequency of the <span class="math notranslate nohighlight">\(A_1\)</span> allele,
<span class="math notranslate nohighlight">\(p\)</span>, is found in the following way</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
    p = (0.25 \times 1) + (0.5 \times\frac{1}{2}) + (0.25\times0) = 0.5.
\end{aligned}\]</div>
<p>So in this trivial example <span class="math notranslate nohighlight">\(p = q = 0.5\)</span>.</p>
<p>Let’s do a slightly less contrived example. Consider the following table
of genotypes taken from screening African school children for the sickle
cell anemia locus. In this case recall that <span class="math notranslate nohighlight">\(A_1A_1\)</span> individuals are
healthy, <span class="math notranslate nohighlight">\(A_2A_2\)</span> individuals are sick, and heterozygotes show increased
resistance to malaria.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Observed Numbers:</p></td>
<td><p>411</p></td>
<td><p>1404</p></td>
<td><p>185</p></td>
</tr>
<tr class="row-odd"><td><p>Relative Frequencies:</p></td>
<td><p>0.2055</p></td>
<td><p>0.702</p></td>
<td><p>0.0925</p></td>
</tr>
</tbody>
</table>
<p>First double check that I have gotten relative genotypic frequencies
correct. Next calculate the allele frequencies. I get <span class="math notranslate nohighlight">\(p = 0.556\)</span>, what
do you get?</p>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="the-hardy-weinberg-law">
<h1>The Hardy-Weinberg Law<a class="headerlink" href="#the-hardy-weinberg-law" title="Permalink to this headline">¶</a></h1>
<p>Perhaps the best known population genetic model is the famous
Hardy-Weinberg law, which describes the relationship between allele and
genotype frequencies at an autosomal locus in an equilibrium randomly
mating population. By equilibrium we mean a population that is <em>not</em>
undergoing evolutionary forces such as selection, mutation, or genetic
drift. By randomly mating we mean that individuals mate with each other
irrespective of relatedness or geography. Populations where cousins mate
are not randomly mating- this is called inbreeding. Populations where
<span class="math notranslate nohighlight">\(A_1A_1\)</span> individuals are more likely to mate with other <span class="math notranslate nohighlight">\(A_1A_1\)</span>
individuals are not randomly mating- we call this assortative mating.
Similarly if one is more likely to mate with their neighbors,
geographically, than this too is not random mating.</p>
<p>The easiest way to envision an honest to goodness Hardy-Weinberg
population is as hermaphrodite, broadcast spawners, something like
sponges or corals. Individuals produce both sperm and eggs, and randomly
let them go into the environment to meet up and form zygotes. In this
way the probability of forming an <span class="math notranslate nohighlight">\(A_1A_1\)</span> zygote is the product of the
probability of choosing an <span class="math notranslate nohighlight">\(A_1\)</span> egg, <span class="math notranslate nohighlight">\(p\)</span>, and the probability of
choosing an <span class="math notranslate nohighlight">\(A_1\)</span> sperm (these probabilities are identical because of
our assumption of hermaphrodites). Thus the probability of randomly
forming an <span class="math notranslate nohighlight">\(A_1A_1\)</span> zygote is <span class="math notranslate nohighlight">\(p^2\)</span>. By similar reasoning, the
probability of forming an <span class="math notranslate nohighlight">\(A_2A_2\)</span> zygote is <span class="math notranslate nohighlight">\(q^2\)</span>. There are two ways
to form an <span class="math notranslate nohighlight">\(A_1A_2\)</span> heterozygote. The first way is with an <span class="math notranslate nohighlight">\(A_1\)</span> sperm
and an <span class="math notranslate nohighlight">\(A_2\)</span> egg. This probability is <span class="math notranslate nohighlight">\(pq\)</span>. The second way would be with
an <span class="math notranslate nohighlight">\(A_2\)</span> sperm and an <span class="math notranslate nohighlight">\(A_1\)</span> egg. Again, this probability is <span class="math notranslate nohighlight">\(pq\)</span>.
Because we sum mutually exclusive probabilities, the total probability
of randomly forming a heterozygote is <span class="math notranslate nohighlight">\(2pq\)</span>. Congratulations, you’ve
just derived your first population genetic model.</p>
<p>After one round of random mating the frequencies of the three genotypes
at our locus are</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>H-W Frequency:</p></td>
<td><p><span class="math notranslate nohighlight">\(p^2\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(2pq\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(q^2\)</span></p></td>
</tr>
</tbody>
</table>
<p>These are the Hardy-Weinberg genotype frequencies. They only depend on
allele frequencies, thus if you know <span class="math notranslate nohighlight">\(p\)</span> you know all of the genotype
frequencies.</p>
<p>A couple of things to note about random mating in our diploid
hermaphrodites:</p>
<ol class="simple">
<li><p>Allele frequencies are unaffected by random mating. Random mating can
change genotype frequencies, but not allele frequencies, thus after one
round of mating Hardy-Weinberg genotype frequencies will remain
unchanged, forever!</p></li>
<li><p>In our diploid hermaphrodites it only takes one generation to
achieve H-W equilibrium proportions. In species with separate sexes it
takes two generations.</p></li>
<li><p>To come up with genotype frequencies after random mating we only
need to know allele frequencies before random mating, not genotype
frequencies.</p></li>
</ol>
<p>Let’s step back to our numerical example from human sickle cell anemia.
Our goal will be to predict Hardy-Weinberg expected values for genotype
frequencies at this locus, given our observed allele frequencies. We
have already done the first step which is to calculate the frequency of
the <span class="math notranslate nohighlight">\(A_1\)</span> allele, in this case <span class="math notranslate nohighlight">\(p = 0.556\)</span>. Now lets calculate the
frequency of the <span class="math notranslate nohighlight">\(A_2\)</span> allele, <span class="math notranslate nohighlight">\(q = 1 - p = 1 - 0.556 = 0.444\)</span>. Using
these allele frequencies we can fill out a table of genotype frequencies</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Genotype:</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_1\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_1A_2\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(A_2A_2\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>H-W Frequency:</p></td>
<td><p><span class="math notranslate nohighlight">\(p^2\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(2pq\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(q^2\)</span></p></td>
</tr>
<tr class="row-odd"><td><p>H-W Expected Frequencies:</p></td>
<td><p>0.309</p></td>
<td><p>0.494</p></td>
<td><p>0.197</p></td>
</tr>
<tr class="row-even"><td><p>Observed Frequencies:</p></td>
<td><p>0.2055</p></td>
<td><p>0.702</p></td>
<td><p>0.0925</p></td>
</tr>
</tbody>
</table>
<p>Compare the observed genotypic frequencies with what we would expect
from the Hardy-Weinberg law. Pretty different, huh? In particular I want
you to notice how we observe many more heterozygotes than we would
expect. This is what is called in the business an excess of
heterozygotes. Why are we seeing this difference? Which of our
Hardy-Weinberg assumptions have failed us? One can always test for a
significant deviation from our H-W expectation using a chi-square
statistic.</p>
<p>So what happens in species with separate sexes (dioecious species)? What
if genotype frequencies are different between the sexes? At the most
extreme, lets imagine that all males are <span class="math notranslate nohighlight">\(A_1A_1\)</span> and all females are
<span class="math notranslate nohighlight">\(A_2A_2\)</span>. With an equal sex ratio the frequency of the <span class="math notranslate nohighlight">\(A_1\)</span> allele is
<span class="math notranslate nohighlight">\(p=0.5\)</span>. In this case, after one round of random mating every individual
in the population is <span class="math notranslate nohighlight">\(A_1A_2\)</span>, and the frequencies of the heterozygotes
is one. This is clearly far from the Hardy-Weinberg expected genotype
frequencies. However, after just one more round of random mating the
frequencies of the <span class="math notranslate nohighlight">\(A_1A_1\)</span>, <span class="math notranslate nohighlight">\(A_1A_2\)</span>, and <span class="math notranslate nohighlight">\(A_2A_2\)</span> genotypes are
returned to their Hardy-Weinberg values of <span class="math notranslate nohighlight">\(\frac{1}{4},\frac{1}{2}\)</span>,
and <span class="math notranslate nohighlight">\(\frac{1}{4}\)</span>. So even for dioecious species with very different
allele frequencies it only takes two rounds of random mating to get back
our H-W expected genotype frequencies.</p>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="heterozygosity">
<h1>Heterozygosity<a class="headerlink" href="#heterozygosity" title="Permalink to this headline">¶</a></h1>
<p>As we talked about in the last lecture, getting a handle on the amounts
of genetic variation within a population is very important both to
genetics and evolutionary theory. One broad way to describe the
“amount” of variation that exists at the genetic level is to describe a
population by the probability of sampling a heterozygous genotype from
it. For obvious reasons we call this measure the heterozygosity of a
population.</p>
<p>Our cursory look at the Hardy-Weinberg law suggests that heterozygosity
might depend on allele frequencies. Figure 1 shows a plot of the
frequency of heterozygotes as a function of the <span class="math notranslate nohighlight">\(A_1\)</span> allele frequency
under Hardy-Weinberg conditions. Ask yourself this simple question- at
what allele frequency is the probability of sampling a heterozygote from
a population maximized? Does this make intuitive sense to you?</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="k">def</span> <span class="nf">het</span><span class="p">(</span><span class="n">p</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Calculate the expected heterozygosity of a population</span>
<span class="sd">    from the allele frequency, p</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="mi">2</span> <span class="o">*</span> <span class="n">p</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">))</span>

<span class="n">allele_freqs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">)</span>
<span class="n">expected_het</span> <span class="o">=</span> <span class="n">het</span><span class="p">(</span><span class="n">allele_freqs</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">allele_freqs</span><span class="p">,</span> <span class="n">expected_het</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;expected heterozygosity&quot;</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;allele frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;expected heterozygosity&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0, 0.5, &#39;expected heterozygosity&#39;)
</pre></div>
</div>
<img alt="_images/lecture1_7_1.png" src="_images/lecture1_7_1.png" />
</div>
</div>
<p>Before moving on to a formal treatment of heterozygosity, let’s first
generalize the Hardy-Weinberg law to the case where we have more than
two alleles at a locus. Good examples of such loci in actual genomes
would include microsatellites or allozyme polymorphisms, as opposed to
SNPs which tend to be diallelic. If we have <span class="math notranslate nohighlight">\(k\)</span> alleles at a locus,
<span class="math notranslate nohighlight">\(A_i\)</span>, <span class="math notranslate nohighlight">\(i=1...k\)</span>, let their frequencies be called <span class="math notranslate nohighlight">\(p_i\)</span>, <span class="math notranslate nohighlight">\(i=1...k\)</span>. It
follows that the frequency of the <span class="math notranslate nohighlight">\(A_iA_i\)</span> homozygote after random
mating should be <span class="math notranslate nohighlight">\(p_{i}^2\)</span> and the frequency of the <span class="math notranslate nohighlight">\(A_iA_j\)</span>
heterozygote should be <span class="math notranslate nohighlight">\(2p_ip_j\)</span>. Simple enough, but make sure this is
straight in your head. It might be helpful to write out the three allele
example in its entirety at this point. Moving on, the total frequency of
homozygotes in our population is</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
G = \sum_{i=i}^{k}p_{i}^2.
\end{aligned}\]</div>
<p><span class="math notranslate nohighlight">\(G\)</span> here is called the
homozygosity of a locus. Ask yourself- what is the probabilistic
interpretation of <span class="math notranslate nohighlight">\(G\)</span>? The heterozygosity of a locus is the complement (in
probability space) of homozygosity</p>
<div class="math notranslate nohighlight">
\[\begin{aligned}
H = 1 - G = 1 - \sum_{i=i}^{k}p_{i}^2.
\end{aligned}\]</div>
<p>For populations
that obey the assumptions of the Hardy-Weinberg law, heterozygosity will
actually equal the frequency of heterozygotes, however heterozygosity is
calculated using only allele frequencies. Because of this heterozygosity
is often used to describe populations that don’t conform to H-W
assumptions. In many ways it’s just a convenient measure- like
temperature.</p>
<div class="section" id="simulating-hardy-weinberg-expectations">
<h2>Simulating Hardy-Weinberg Expectations<a class="headerlink" href="#simulating-hardy-weinberg-expectations" title="Permalink to this headline">¶</a></h2>
<p>Next let’s use SLiM to simulate a finite population that is randomly mating. We will add a single mutation to our population at the beginning at a specific frequency, <span class="math notranslate nohighlight">\(p_0\)</span>,
and then watch how the genotype frequencies change over time.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> slim_magic

<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span><span class="p">,</span> <span class="n">SVG</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">slim_stats_reps_rstack</span> 10 --out df
// set up a single locus simulation of drift
initialize()
{
    // set the overall mutation rate
    // no mutation in this simulation
    initializeMutationRate(0);
    // m1 mutation type: neutral
    initializeMutationType(&quot;m1&quot;, 0.5, &quot;f&quot;, 0.0);
    // single locus simulation
    // g1 genomic element type: uses m1 probability 1
    initializeGenomicElementType(&quot;g1&quot;, c(m1), c(1.0));
    // uniform chromosome of length 1 site
    initializeGenomicElement(g1, 0, 0);
    // uniform recombination along the chromosome
    initializeRecombinationRate(0);
    suppressWarnings(T);
}

// create a population of 100 individuals
1 {
    sim.addSubpop(&quot;p1&quot;, 100);
    // sample 100 haploid genomes 
    target = sample(p1.genomes, 100);
    // add a mutation to those genomes
    // H_0 = 0.5 here
    target.addNewMutation(m1,0, 0);
    cat(&quot;generation,p,x11,x12,x22\\n&quot;);
}
1:300 late(){
    inds = p1.sampleIndividuals(100);
    // record the number of mutations in each individual
    ind_count = inds.countOfMutationsOfType(m1);
    counts = c(0, 0, 0);
    for (x in ind_count)
        counts[x] = counts[x] + 1;
    // divide by popn size to get genotype freqs
    counts = counts / 100;
    // allele freq of A_1 allele in current gen
    freqs = sim.mutationFrequencies(p1);
    // each gen print out the gen, the allele freqs, and the counts
    if (length(freqs) &gt; 0.0)
        catn(sim.generation + &quot;,&quot; + freqs + &quot;,&quot; + paste(counts, sep=&quot;,&quot;));
    }
// run to generation 0
300 late() {
    sim.simulationFinished();
    }
</pre></div>
</div>
</div>
</div>
<p>Next let’s write a tiny function that will take an allele frequency and return the <em>expected</em> HW genotype frequencies</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">hwe</span><span class="p">(</span><span class="n">p</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">p</span><span class="o">**</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span> <span class="o">*</span> <span class="n">p</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">),</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">)</span><span class="o">**</span><span class="mi">2</span><span class="p">])</span>

<span class="c1">#run that function for 100 pts between (0,1)</span>
<span class="n">expected</span> <span class="o">=</span> <span class="n">hwe</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<p>Finally let’s plot for our simulations how the observed genotype frequencies compare to that expected, given the
observed allele frequency</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#plot simulated</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">df</span><span class="o">.</span><span class="n">x11</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_11 frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">df</span><span class="o">.</span><span class="n">x12</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_12 frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">p</span><span class="p">,</span> <span class="n">df</span><span class="o">.</span><span class="n">x22</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;A_22 frequency&quot;</span><span class="p">)</span>


<span class="c1">#plot expected</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">0</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;HWE expected&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">1</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">100</span><span class="p">),</span><span class="n">expected</span><span class="p">[</span><span class="mi">2</span><span class="p">,:],</span> <span class="n">c</span><span class="o">=</span><span class="s2">&quot;red&quot;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s2">&quot;dotted&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;allele frequency&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;genotype frequency&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0, 0.5, &#39;genotype frequency&#39;)
</pre></div>
</div>
<img alt="_images/lecture1_16_1.png" src="_images/lecture1_16_1.png" />
</div>
</div>
<p>Wow an awesome fit! Why does this look so good?</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="intro.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Welcome to Population Genetics</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="lecture2.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Natural Selection</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Andrew Kern and Peter Ralph<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>